apps/examples/webserver_test/webserver_test_util.c

```diff
@@ -0,0 +1,261 @@
+		printf("Error: buffer is too small");
+		return NULL;
+	}
+
+	memcpy(dest, src, len);
+	dest[len] = '\0';
+	return dest + len;
+}
+
+int find_first_crlf(const char *src, int len, int start)
+{
+	int i = 0;
+
+	for (i = start + 1; i < len; i++) {
+		/* Check CR and LF */
+		if (src[i] == '\n' && src[i - 1] == '\r') {
+			return i - 1;
+		}
+	}
+
+	return -1;
+}
+
+int separate_status_line(const char *src, int *httpver, int *status, char *phrase)
+{
+	int i = 0;
+	int divide[6] = {0, };
+	int divide_pos = 0;
+
+	int line_length = 0;
+	char status_code[4] = {0, };
+	int status_start = 0;
+	int phrase_start = 0;
+	int phrase_length = 0;
+
+	*httpver = HTTP_HTTP_VERSION_UNKNOWN;
+	phrase[0] = '\0';
+
+	line_length = strlen(src);
+
+	/* Get divide positions */
+	for (i = 0; i < line_length; i++) {
+		if (src[i] == ' ') {
+			divide[divide_pos++] = i;
+
+			if (divide_pos >= (sizeof(divide) / sizeof(divide[0]))) {
+				printf("Error: Not HTTP status line!!\n");
+				return WS_TEST_ERR;
+			}
+		}
+	}
+
+	/* Get http version */
+	if (strncmp(src, "HTTP/1.1", 8) == 0) {
+		*httpver = HTTP_HTTP_VERSION_11;
+	} else if (strncmp(src, "HTTP/1.0", 8) == 0) {
+		*httpver = HTTP_HTTP_VERSION_10;
+	} else if (strncmp(src, "HTTP/0.9", 8) == 0) {
+		*httpver = HTTP_HTTP_VERSION_09;
+	}
+
+	/* Get status code */
+	status_start = divide[0] + 1;
+	strncpy(status_code, src + status_start, 4);
+	*status = atoi(status_code);
+
+	/* Get reason phrase */
+	phrase_start = divide[1] + 1;
+	phrase_length = line_length - phrase_start + 1;
+
+	strncpy(phrase, src + phrase_start, phrase_length);
+
+	return WS_TEST_OK;
+}
+
+int separate_keyvalue(const char *src, char *key, char *value)
+{
+	int i = 0;
+	int value_position = 0;
+	int src_len = strlen(src);
+
+	for (i = 0; i < src_len; i++) {
+		if (i >= HTTP_CONF_MAX_KEY_LENGTH) {
+			printf("Error: The key length is over the value buffer size.\n");
+			return WS_TEST_ERR;
+		}
+		if (src[i] != ':') {
+			key[i] = src[i];
+		} else {
+			value_position = i + 2; /* 2 because at the front of value, there is a space */
+			key[i] = '\0';
+			break;
+		}
+	}
+
+	for (i = value_position; i < src_len; i++) {
+		if ((i - value_position) >= HTTP_CONF_MAX_VALUE_LENGTH) {
+			printf("Error: The value length is over the value buffer size.\n");
+			return WS_TEST_ERR;
+		}
+		value[i - value_position] = src[i];
+	}
+	value[i - value_position] = '\0';
+
+	return WS_TEST_OK;
+}
+
+int parse_message(char *buf, int buf_len, char *url,
+					   char **body, int *enc, int *state,
+					   struct http_message_len_t *len,
+					   struct http_client_response_t *response)
+{
+	int protocol = 0;
+	int sentence_end = 0;
+	char key[HTTP_CONF_MAX_KEY_LENGTH] = { 0, };
+	char value[HTTP_CONF_MAX_VALUE_LENGTH] = { 0, };
+	int process_finish = false;
+	int read_finish = false;
+
+	while (!process_finish) {
+		/* At this point, we read a line of http request */
+		switch (*state) {
+		case HTTP_REQUEST_HEADER:
```

Comment:
How about to modify this code like following to increase readability
```
while () {
  case HTTP_REQUEST_HEADER:
    _parse_header();
    break;
  case HTTP_REQUEST_PARAMETERS:
    _parse_parameter();
    break;
...
}
```

Reply:
done